### ISSUE TITLE
Service needs to be restarted in order to detect new bluetooth adapter.

### ISSUE THREAD

**Issue by MHDante** [cboulay/PSMoveService#339]
Can we trigger detection of new bluetooth device from the config tool? If not,, can we trigger service restart from the config tool?

**Comment by cboulay** [comment]
Either of these solutions will be a lot of work, and I suspect there's a very limited use case for it. Can you explain why you need to frequently unplug/replug your bluetooth dongles?

**Comment by MHDante** [comment]
We're in a development lab where students have access to VR hardware to use on computers with pre-installed software. In order to keep our DK2s usable with our labs (which use Vives), we've started using PSMoveService to make the programming easier. (Through OpenVR). The issue is that PSMoveService requires Admin permissions to run, so I've installed it as a service using NSSM. 

Often, what this means is that the users will plug in the hardware after the computer has been started, and have a hard time restarting the service without some Service permissions magic.

**Comment by cboulay** [comment]
You got me. I can't think of a simple way around that. I don't have time to work on that and it's probably way down the list of priorities for @HipsterSloth . If you have someone on your end that could code that in then we'd be happy to guide them and merge a pull request.

**Comment by MHDante** [comment]
I'm currently trying to get the repo compiling locally and will take a stab at it myself. Rather than going towards the bluetooth re-initialization, I'm looking at how to allow user (non-Admin) accounts to restart the service. Perhaps something like what is seen in #45, but with a [Service Controller](https://msdn.microsoft.com/en-us/library/system.serviceprocess.servicecontroller(v=vs.110).aspx?cs-save-lang=1&cs-lang=cpp), preferably in C++.

**Comment by HipsterSloth** [comment]
Oh! Sorry about that complexity. I think the only reason I have PSMoveService set to be an admin executable is so that it can install itself as a window service using the Win32 CreateService api.  I think we could migrated the service installation to a separate executable. I've been looking for an excuse to rip out the boost::application code from psmoveservice anyway.

**Comment by MHDante** [comment]
Yeah, that's what I was just seeing. Thing is, in last year's course, the only part of PSMove-Unity5 that required admin rights was when using Pair-Move-Win. I was wondering if PSMoveService was still writing to the registry (It still does, from what I can tell), and if it required admin rights for that.

**Comment by HipsterSloth** [comment]
Ah right. Yeah it does in fact still write to the registry during pairing here:

https://github.com/cboulay/PSMoveService/blob/master/src/psmoveservice/Platform/BluetoothRequestsWin32.cpp#L970-L1023

That was originally added for Win8 support. Now that were not really supporting versions of windows before Win10 it might be interesting to try removing the registry patching to see if it's still needed.
